tests/greendns_test.py

```diff
@@ -854,3 +857,28 @@ def test_hosts_priority():
 def test_import_rdtypes_then_eventlet():
     # https://github.com/eventlet/eventlet/issues/479
     tests.run_isolated('greendns_import_rdtypes_then_eventlet.py')
+
+
+def test_raise_dns_tcp():
+    # https://github.com/eventlet/eventlet/issues/499
+    try:
+        dns.resolver.query('host.example.com', 'a', tcp=True)
+        assert False, "Should have arisen NXDOMAIN or NoNameservers"
+    except (dns.resolver.NXDOMAIN, dns.resolver.NoNameservers):
+        pass
+
+
+def test_noraise_dns_tcp():
```

Comment:
- `TinyDNSServerTCP` listens `localhost` (and there is no way to change it and that's fine)
- test goes to `127.0.0.1` which is not the same as localhost
- `127...` string duplicate
- `192.168.1.1` is not used
- response is not checked against `192...`